airbyte-integrations/connectors/source-datascope/source_datascope/source.py

```diff
@@ -0,0 +1,160 @@
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+from datetime import datetime, timedelta
+import logging
+import backoff
+
+import requests
+from requests.exceptions import ConnectionError
+from requests.structures import CaseInsensitiveDict
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.sources.streams.http.auth import NoAuth
+
+
+
+# Basic full refresh stream
+class DatascopeStream(HttpStream, ABC):
+
+    # TODO: Fill in the url base. Required.
+    url_base = "https://mydatascope.com/api/external/"
+
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+    @backoff.on_exception(backoff.expo, requests.exceptions.ConnectionError, max_tries=7)
+    def _request(self, url: str, method: str = "GET", data: dict = None) -> List[dict]:
```

Comment:
no need to override `_request`, you need to: 

```
def parse_response(self, response): 
    return [response.json()] if isinstance(response, list) else response.json()

def request_headers(self, **kwargs):
    return {} # fill in your headers here
```

Reply:
I'm using for retrieving the schema dynamically, how I can achieve that without own _request method?

Reply:
you can override `get_json_schema` as described [here](https://docs.airbyte.io/contributing-to-airbyte/python/concepts/schemas)

Reply:
Sorry I don’t understand well what I need to improve here, I used this that was recommended to me to manage dynamic schemas: https://github.com/airbytehq/airbyte/pull/3911/files#diff-bc8fbe92b8635b5585d5137dc0ac85d4634e2681032396f978b16fcafa6b8e44R154  I'm already using get_json _schema but retrieving data using this _request method inside the get_json _schema. It's that ok? Or I need to use another request method the get that metadata?